zerver/lib/kroki.py

```diff
@@ -0,0 +1,40 @@
+from typing import Optional
+
+import requests
+from django.conf import settings
+
+
+def render_kroki(text: str, diagram_type: str) -> Optional[str]:
+    """Render diagrams from textual descriptions
+
+    Returns the HTML string, or None in cases like there was some
+    syntax error or diagram type was not correct.
+
+    Keyword arguments:
+    text -- Text string with the Kroki diagram to render.
+            Don't split text into paragraphs as they often
+            contain information which links each other.
+
+    diagram_type -- Text string required when making the POST request.
+                    This indicates the diagram type we want for the
+                    given input source.
+    """
+
+    if not settings.KROKI_SERVER_URL:
+        return None
+
+    data = dict(
+        diagram_source=text,
+        diagram_type=diagram_type,
+        output_format="SVG",
+    )
```

Comment:
So I'm worried about the security implications of including arbitrary SVGs in a Zulip message.  I wonder if we want to make this integration instead upload the returned SVG as a Zulip file upload and then link to it like any other inline image, rather than putting it into our message content.

@andersk thoughts?  That process has somewhat involved infrastructure and is messy for message editing, so if there's a better way to restrict the XSS risk here, that'd be preferable.